### ISSUE TITLE
[🐛 Bug]: Using CDP on Firefox makes the grid unstable

### ISSUE THREAD

**Issue by mythsunwind** [SeleniumHQ/selenium#11083]
### What happened?

Unfortunately I'm currently not really sure where the issue is located. Could be geckodriver, selenium grid, selenium node, docker-selenium.

We are able to get developer console logs via the following code:
```
WebDriver driver = new Augmenter().augment(this.driver);
                    DevTools devTools = ((HasDevTools) driver).getDevTools();
                    devTools.createSession();
                    // Disable logger for selenium package
                    Logger.getLogger("org.openqa.selenium").setLevel(Level.OFF);
                    // Override stderr to somehow bring disabled logger in effect (no clue why this is needed)
                    ByteArrayOutputStream baos = new ByteArrayOutputStream();
                    PrintStream newStdErr = new PrintStream(baos);
                    System.setErr(newStdErr);

                    devTools.getDomains().events().addConsoleListener(
                            event -> {
                                logManager.addEntry(new LogEntry(
                                                logManager.toLevel(event.getType()),
                                                // Adding offset for time zone
                                                event.getTimestamp().getEpochSecond(),
                                                event.getArgs().toString()
                                        )
                                );
                            }
                    );
```
This creates a lot of exceptions on the java side which we only can partly disable by overriding the logger.
You can see above that we additionally needed to reset stderr to not flood our test output with these exception output that seems to come directly on stderr and is not catchable.

One issue remains: The grid becomes quite unstable with the above code and usually breaks after 3-4 days. We have seens two different behaviours:
* selenium grid hub is still reachable, but shows "loading..." for all grid nodes (all docker-selenium containers are all still marked healthy)
* selenium grid hub not reachable anymore (docker-selenium containers for the grid hub are unhealthy)

We currently run:
* docker-selenium [4.4.0-20220831](https://github.com/SeleniumHQ/docker-selenium/releases/tag/4.4.0-20220831)
* selenium-java is on version 4.1.1 (we unfortunately cannot update to 4.4.0 due to dependency issues)

### How can we reproduce the issue?

```shell
This is not easy to reproduce. We have test runs with over 1200 tests in one suite which run usually 2 times a day. Only after 2-5 days it breaks.
```

### Relevant log output

```shell
Will add logs once it happens again
```

### Operating System

debian

### Selenium version

Java 4.1.1

### What are the browser(s) and version(s) where you see this issue?

Firefox 105

### What are the browser driver(s) and version(s) where you see this issue?

GeckoDriver: 0.31.0

### Are you using Selenium Grid?

docker-selenium 4.4.0-20220831

**Comment by pujagani** [comment]
Thank you for the details. This is a tricky one since there is no proper way to reproduce it.
Have you only observed this behavior with CDP on Firefox? Does it happen with CDP using ChromeDriver? Does it happen if CDP is not used at all? 

```
This creates a lot of exceptions on the java side which we only can partly disable by overriding the logger.
You can see above that we additionally needed to reset stderr to not flood our test output with these exception output that seems to come directly on stderr and is not catchable.
```
Can you elaborate on the concern here? Can you share a complete reproducible test snippet that leads to the exceptions? 
Firefox CDP is not complete implementation-wise from the Firefox side. They have implemented only a handful of CDP domains. I am wondering if that causing the exceptions or something on Selenium side is causing it.

**Comment by pujagani** [comment]
```
selenium grid hub is still reachable, but shows "loading..." for all grid nodes (all docker-selenium containers are all still marked healthy)
selenium grid hub not reachable anymore (docker-selenium containers for the grid hub are unhealthy)"

```
To add to this, similar errors in terms of showing loading state and Grid being stuck were fixed thanks to Diego's work https://github.com/SeleniumHQ/selenium/commit/f569d7f6b1f2a35ea41d9604cdc69dbbd5f58485 . The cause of the error was different but it caused the "loading.." state of the GridUI. The changes will be available as part of the release. I am hoping the fix helps for this scenario as well.

**Comment by titusfortner** [comment]
Is this fixed in 4.6?

**Comment by pujagani** [comment]
We were not able to reproduce this exactly with CDP. `selenium grid hub is still reachable, but shows "loading..." for all grid nodes (all docker-selenium containers are all still marked healthy)` - this bit was fixed in 4.6. But if something else is causing it then we can reopen the issue.
@mythsunwind Can you please try with 4.6 and let us know? Thank you!

**Comment by diemol** [comment]
I will close this due to the low activity but always happy to revisit if needed.
